File: frontend/src/api/loginHistory.ts
import api from './index'

export interface LoginHistory {
  id: number
  username: string
  client_ip: string
  success: boolean
  created_at: string
}

export const listLoginHistory = (limit = 500) =>
  api.get<LoginHistory[]>('/admin/login-history', { params: { limit } })

File: store/login_history.go
package store

func CreateLoginHistory(username, clientIP string, success bool) error {
	return DB.Create(&LoginHistory{Username: username, ClientIP: clientIP, Success: success}).Error
}

func ListLoginHistory(limit int) ([]LoginHistory, error) {
	var records []LoginHistory
	err := DB.Order("created_at DESC").Limit(limit).Find(&records).Error
	return records, err
}
